apps/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -292,7 +285,11 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
           throw new AlreadyReportedError();
         }
 
-        if (this.getShrinkwrapHash() !== policyOptions.repoState.pnpmShrinkwrapHash) {
+        const hashSerializeOptions: IPnpmShrinkWrapFileSerializeOptions = {
+          omitImporters: policyOptions.validateOnlyExternalPackageLayout
```

Comment:
It feels a little odd that this is set two different ways in the two places the `getShrinkwrapHash` function is called. Do we not have access to the `policyOptions` object in the `refreshState` function?

Reply:
There isn't really a common node between these two code paths. If I pass it into RepoStateFile during construction, that's a public API change, which so far this experiment has avoided doing.

Reply:
Found a solution. Passing `IExperimentsJson` as an optional 3rd parameter to the validate() function, and now all the code paths in `PnpmShrinkwrapFile` just use the `IExperimentsJson` for consistency (`getShrinkwrapHash()` and `validate()`).